skyportal/handlers/api/source.py

```diff
@@ -707,6 +716,14 @@ class Validator(Schema):
             q = q.filter(Source.saved_at <= saved_before)
         if saved_after:
             q = q.filter(Source.saved_at >= saved_after)
+        if list_name:
+            obj_list = Listing.query.filter(
+                Listing.list_name == list_name,
+                Listing.user_id == self.associated_user_object.id,
+            ).all()
+            obj_id_list = [obj.obj_id for obj in obj_list]
+            if obj_id_list is not None:
+                q = q.filter(Source.obj_id.in_(obj_id_list))
```

Comment:
I believe this will not use the db indices as well as if you used joins. There's a concept called "sargable" which describes which sorts of queries best use indices - basically, I think using "IN" is less optimal than using joins and WHERE statements (filter with SA). I would suggest doing something like the classifications where you join in `Listing` earlier in the handler and then filtering without the `in_`.

Reply:
OK it looks a lot better now. I still don't see a big change in the responsiveness of the button. 
On the other hand, I only have one list and two sources... 
So I think something is going on in the JS side that takes too long until I get the favorites list in the state tree.